tutorials/migration.md

```diff
@@ -4,96 +4,846 @@ layout: intro
+installed.
+
+`feature:repo-list`
+
+![result of running feature:repo-list](images/feature-list.png)
+
+The following command will give the list of available add-ons.
+
+`feature:list`
+
+![result of running feature:list](images/feature-available.png)
+
+For each add-on you are using in openHAB 1.x, and for each transformation type
+you are using, install the corresponding add-on using:
+
+`feature:install <add-on name>`
+
+using the name from the list. For example, to install the Weather Binding run:
+
+`feature:install openhab-binding-weather`
+
+To get a list of all currently installed items run:
+
+`feature:list | grep Started`
+
+At this time only install add-ons that are 1.9.0 SNAPSHOT. The 2.0 add-ons will 
+be installed later.
+
+Note: if running in Docker, create `<openHAB home>/userdata/persistence` manually prior to 
+installing persistence add-ons.
+
+###PaperUI Approach
+
+These instructions assume no edits have been made to `runtime.cfg` or 
+`addons.cfg` as described in the previous section.
+
+Navigate your browser to `http://<openhab server address>:8080` and select 
+PaperUI from the list of UIs. Navigate to `Configuration -> System` on the left
+hand side and enable "Include Legacy 1.x Bindings" and "Access Remote 
+Repositories". This will allow you to install your current bindings even if 
+there is an openHAB 2 version available. Press "Save".
+
+![PaperUI System Settings](images/PaperUI-Config-System.png)
+
+Now select "Extensions" from the left and browse or search for each of the 
+bindings and add-ons you currently use. Make note of those that do not appear in
+the list. Do not install any binding that has a version 2.0. If there is not a
+1.9.0 version we will install the binding manually (see below).
+
+This is a great opportunity to identify those bindings you actually use and
+only install those.
+
+When you find a binding to install, press the Install button and wait for 
+installation to complete. Do not install the My.openHAB Binding yet.
+
+If you are running openHAB in a Docker container you need to create 
+`<openHAB home>/userdata/persistence` manually prior to installing a persistence 
+add-on.
+
+One thing to be aware of is any configuration done in PaperUI or Habmin gets 
+saved to a non-human readable database. And since none of the administration 
+UIs are currently capable of doing everything, you will end up with a mix of 
+database and text based configuration. This is why we recommend not using 
+PaperUI for those migrating to openHAB 2 from openHAB 1.x.
+
+###Configure Add-ons
+
+As bindings and other add-ons are installed, you can watch for errors in the 
+logs. Errors at this time may not be important but make note of those that did
+generate errors during installation for special attention later.
+
+Once a binding or add-on is installed, it will create a 
+`conf/services/<add-on name>.cfg` file. Unlike in openHAB 1.x where all the 
```

Comment:
...for manual installations, since apt packages are /etc/openhab2/services/<add-on name>.cfg